src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -12,37 +14,194 @@ internal static partial class ClrThreadPool
         
         private static int s_minThreads; // TODO: Initialize
         private static int s_maxThreads; // TODO: Initialize
+        private static readonly LowLevelLock s_maxMinThreadLock = new LowLevelLock();
+        
+        private static readonly LowLevelLock s_threadAdjustmentLock = new LowLevelLock();
+        private static ThreadCounts s_counts = new ThreadCounts();
+        private static int s_lastDequeueTime;
+        private static int s_priorCompletedWorkRequestsTime;
+        private static int s_nextCompletedWorkRequestsTime;
+        private static long s_currentSampleStartTime;
+        private static int s_priorCompletionCount = 0;
```

Comment:
I've attempted to add code to ensure cache alignment in [04d10343b](https://github.com/dotnet/corert/pull/3843/commits/04d10343b5f3c68d23710ee8ac48ee469ef6d79b). Can you check over it when you have a chance to make sure that it actually works correctly?